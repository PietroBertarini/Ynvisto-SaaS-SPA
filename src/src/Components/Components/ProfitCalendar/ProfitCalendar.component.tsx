import React from 'react';
import Grid from '@material-ui/core/Grid';
import ProfitCard, { ProfitCardProps } from '../ProfitCard/ProfitCard.component';
import { useStyles } from './ProfitCalendar.styles';

export interface OwnProps {
    /**
     * Array of api that contain all data from profit and tax of one year.
     */
    monthArray : ProfitCardProps[];
}

/**
 * Calendar of profits and tax
 * @constructor
 */
function ProfitCalendar(props: OwnProps) {
  const {
    monthArray,
  } = props;

  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid container item xs={12} spacing={2}>
          <FormRow threeMonthArray={monthArray.slice(0, 3)} />
        </Grid>
        <Grid container item xs={12} spacing={2}>
          <FormRow threeMonthArray={monthArray.slice(3, 6)} />
        </Grid>
        <Grid container item xs={12} spacing={2}>
          <FormRow threeMonthArray={monthArray.slice(6, 9)} />
        </Grid>
        <Grid container item xs={12} spacing={2}>
          <FormRow threeMonthArray={monthArray.slice(9, 12)} />
        </Grid>
      </Grid>
    </div>
  );
}

export interface FormRowProps {
    threeMonthArray : ProfitCardProps[];
}
function FormRow(props: FormRowProps) {
  const {
    threeMonthArray,
  } = props;
  return (
    <>
      <Grid item xs={4}>
        <ProfitCard
          status={threeMonthArray[0].status}
          profit={threeMonthArray[0].profit}
          tax={threeMonthArray[0].tax}
          month={threeMonthArray[0].month}
        />
      </Grid>
      <Grid item xs={4}>
        <ProfitCard
          status={threeMonthArray[1].status}
          profit={threeMonthArray[1].profit}
          tax={threeMonthArray[1].tax}
          month={threeMonthArray[1].month}
        />
      </Grid>
      <Grid item xs={4}>
        <ProfitCard
          status={threeMonthArray[2].status}
          profit={threeMonthArray[2].profit}
          tax={threeMonthArray[2].tax}
          month={threeMonthArray[2].month}
        />
      </Grid>
    </>
  );
}
export default ProfitCalendar;
