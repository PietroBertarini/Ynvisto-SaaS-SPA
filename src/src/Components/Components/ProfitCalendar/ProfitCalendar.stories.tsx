import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import ProfitCalendar, { OwnProps } from './ProfitCalendar.component';
import { mockCalendarData } from '../../../Mocks/ProfitCalendar.mock';

export default {
  title: 'Components/ProfitCalendar',
  component: ProfitCalendar,
} as Meta;

const Template: Story<OwnProps> = (args) => <ProfitCalendar {...args} />;

export const Calendar = Template.bind({});
Calendar.args = {
  monthArray: mockCalendarData,
};
