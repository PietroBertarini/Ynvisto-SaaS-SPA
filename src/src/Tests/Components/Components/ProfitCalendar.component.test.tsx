import React from 'react';
import { shallow } from 'enzyme';
import ProfitCalendar from '../../../Components/Components/ProfitCalendar/ProfitCalendar.component';
import { mockCalendarData } from '../../../Mocks/ProfitCalendar.mock';

test('Renders ProfitCalendar component and match to previously snapshot', () => {
  const component = shallow(<ProfitCalendar monthArray={mockCalendarData} />);
  expect(component).toMatchSnapshot();
});
