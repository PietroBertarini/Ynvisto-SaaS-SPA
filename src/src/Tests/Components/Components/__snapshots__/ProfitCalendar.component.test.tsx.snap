// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ProfitCalendar component and match to previously snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={2}
      xs={12}
    >
      <FormRow
        threeMonthArray={
          Array [
            Object {
              "month": "Janeiro",
              "profit": 30304.94,
              "status": "LATE",
              "tax": 506.14,
            },
            Object {
              "month": "Fevereiro",
              "profit": 30304.94,
              "status": "LATE",
              "tax": 506.14,
            },
            Object {
              "month": "MarÃ§o",
              "profit": 30304.94,
              "status": "PAID",
              "tax": 506.14,
            },
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={2}
      xs={12}
    >
      <FormRow
        threeMonthArray={
          Array [
            Object {
              "month": "Abril",
              "profit": 30304.94,
              "status": "NO_DARF",
              "tax": 506.14,
            },
            Object {
              "month": "Maio",
              "profit": 30304.94,
              "status": "PENDING",
              "tax": 506.14,
            },
            Object {
              "month": "Junho",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={2}
      xs={12}
    >
      <FormRow
        threeMonthArray={
          Array [
            Object {
              "month": "Julho",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
            Object {
              "month": "Agosto",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
            Object {
              "month": "Setembro",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      spacing={2}
      xs={12}
    >
      <FormRow
        threeMonthArray={
          Array [
            Object {
              "month": "Outubro",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
            Object {
              "month": "Novembro",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
            Object {
              "month": "Dezembro",
              "profit": 30304.94,
              "status": "NOT_FINISHED",
              "tax": 506.14,
            },
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
